### PR TITLE
Better error message when no backend engine is found.

### LINKED ISSUE #5291: ds = xr.tutorial.load_dataset("air_temperature") with 0.18 needs engine argument
*From xarray-contrib/xarray-tutorial#43 by @scottyhq*:

Many notebooks out there start with the line `ds = xr.tutorial.load_dataset("air_temperature")`. That now gives an error traceback with xarray>=0.18:

```pytb
Traceback (most recent call last):
  File "/Users/scott/GitHub/zarrdata/./create_zarr.py", line 6, in <module>
    ds = xr.tutorial.load_dataset("air_temperature")
  File "/Users/scott/miniconda3/envs/zarrdata/lib/python3.9/site-packages/xarray/tutorial.py", line 179, in load_dataset
    with open_dataset(*args, **kwargs) as ds:
  File "/Users/scott/miniconda3/envs/zarrdata/lib/python3.9/site-packages/xarray/tutorial.py", line 100, in open_dataset
    ds = _open_dataset(filepath, **kws)
  File "/Users/scott/miniconda3/envs/zarrdata/lib/python3.9/site-packages/xarray/backends/api.py", line 485, in open_dataset
    engine = plugins.guess_engine(filename_or_obj)
  File "/Users/scott/miniconda3/envs/zarrdata/lib/python3.9/site-packages/xarray/backends/plugins.py", line 112, in guess_engine
    raise ValueError("cannot guess the engine, try passing one explicitly")
ValueError: cannot guess the engine, try passing one explicitly
```

It's an easy fix though, just add `ds = xr.tutorial.load_dataset("air_temperature", engine="netcdf4")`, new users might be thrown by that though. Also a note that unless the `netcdf4` library is explicitly put into the software environment, even adding the engine=netcdf4 can result in an error: "ValueError: unrecognized engine netcdf4 must be one of: ['store', 'zarr']", so I think a minimal environment definition to run would be:
```
name: xarray-tutorial
channels:
  - conda-forge
dependencies:
  - xarray=0.18
  - pooch=1.3
  - netcdf4=1.5
  - zarr=2.8
```

### PR DESCRIPTION
**Pull request by shoyer**
Also includes a better error message when loading a tutorial dataset but an underlying IO dependency is not found.

- [x] Fixes #5291
- [x] Tests added
- [x] Passes `pre-commit run --all-files`

### CODE REVIEW

**Comment by shoyer** [comment]
@alexamici @aurghs any thoughts?

File: xarray/tutorial.py [hunk 1]
```diff
@@ -36,13 +36,48 @@ def _construct_cache_dir(path):
     "RGB.byte": "https://github.com/mapbox/rasterio/raw/1.2.1/tests/data/RGB.byte.tif",
     "shade": "https://github.com/mapbox/rasterio/raw/1.2.1/tests/data/shade.tif",
 }
+file_formats = {
+    "air_temperature": 3,
+    "rasm": 3,
+    "ROMS_example": 4,
+    "tiny": 3,
+    "eraint_uvz": 3,
+}
```
Comment (keewis) [inline xarray/tutorial.py:45]: should we also mention the backend in the list of the available datasets for `open_dataset`?

**Comment by andersy005** [comment]
> @alexamici @aurghs any thoughts?

@alexamici @aurghs, the gentlest of bumps on this. Have a few minutes to take a look at this? :smile:

**Comment by aurghs** [comment]
That's perfect.

**Review by aurghs — APPROVED** [review]

**Review by andersy005 — APPROVED** [review]

File: xarray/tutorial.py (hunk 1 above)
Comment (shoyer) [reply xarray/tutorial.py:45]: This would be a good idea for future work :)

**Pull request merged**